### PR TITLE
OSD: Adds a callsign for OSD

### PR DESCRIPTION
**Pull request by tridge**
Uses a file "callsign.txt" on the sdcard. Example with MYNAME in callsign.txt:

image

### CODE REVIEW

**Comment by KenwoodFox** [comment]
This feature is awesome. Best of luck.

File: libraries/AP_OSD/AP_OSD_Screen.cpp [hunk 1]
```diff
@@ -1749,6 +1769,33 @@ void AP_OSD_Screen::draw_pluscode(uint8_t x, uint8_t y)
...
 #endif
 
+/*
+  support callsign display from a file called callsign.txt
+ */
+void AP_OSD_Screen::draw_callsign(uint8_t x, uint8_t y)
+{
+#if HAVE_FILESYSTEM_SUPPORT
+    if (!callsign_data.loaded) {
+        callsign_data.loaded = true;
```
Comment (magicrub) [inline libraries/AP_OSD/AP_OSD_Screen.cpp:1779]: the way you're using this `loaded` variable is misleading. It's always true, even if loading failed. Needs a new name like `init_has_run_once`

**Review by magicrub — CHANGES_REQUESTED** [review]
variable name is misleading. Otherwise, LGTM

File: libraries/AP_OSD/AP_OSD_Screen.cpp (hunk 1 above)
Comment (tridge) [reply libraries/AP_OSD/AP_OSD_Screen.cpp:1779]: fixed

**Comment by magicrub** [comment]
LGTM, thanks Tridge!

**Pull request merged**